vllm/lora/models.py

```diff
@@ -469,20 +472,24 @@ def activate_lora(
                 module.reset_lora(index)
         return True
 
+    @property
+    def activate_adapter(self):
+        return self.activate_lora
+
     def _deactivate_lora(self, lora_id: int):
         try:
             index = self.lora_index_to_id.index(lora_id)
             self.lora_index_to_id[index] = None
         except ValueError:
             pass
 
-    def deactivate_lora(self, lora_id: int) -> bool:
-        """Remove a LoRA from a GPU buffer."""
-        if lora_id in self._active_loras:
-            self._deactivate_lora(lora_id)
-            self._active_loras.pop(lora_id)
-            return True
-        return False
+    @property
+    def _deactivate_adapter(self):
+        return self._deactivate_lora
+
+    @property
+    def deactivate_lora(self):
+        return self.deactivate_adapter
```

Comment:
this is a bit of an odd pattern (property returning a method)

Reply:
@Yard1 The reason for this pattern is that I want to keep the original LoRA functions untouched while also abstracting the common functions in the base adapter classes.
So for instance, the base adapter class has methods like `deactivate_adapter` which is common between both lora and prompt-adapter codebases so the functionality has been written there. But in order to keep the LoRA function `deactivate_lora` I am routing it to the base function.
Any other way we should solve this?

Reply:
I think it's just going to be better to do normal methods and call `super().`